processor/ratelimitprocessor/local.go

```diff
@@ -59,20 +68,44 @@ func (r *localRateLimiter) RateLimit(ctx context.Context, hits int) error {
 	switch r.cfg.ThrottleBehavior {
 	case ThrottleBehaviorError:
 		if ok := limiter.AllowN(time.Now(), hits); !ok {
+			r.requestTelemetry(ctx, []attribute.KeyValue{
+				telemetry.WithReason(telemetry.StatusOverLimit),
+				telemetry.WithDecision("throttled"),
+			})
 			return errTooManyRequests
 		}
 	case ThrottleBehaviorDelay:
-		r := limiter.ReserveN(time.Now(), hits)
-		if !r.OK() {
+		lr := limiter.ReserveN(time.Now(), hits)
+		if !lr.OK() {
+			r.requestTelemetry(ctx, []attribute.KeyValue{
+				telemetry.WithReason(telemetry.TooManyRequests),
+				telemetry.WithDecision("throttled"),
+			})
 			return errTooManyRequests
 		}
-		timer := time.NewTimer(r.Delay())
+		timer := time.NewTimer(lr.Delay())
 		defer timer.Stop()
 		select {
 		case <-ctx.Done():
 			return ctx.Err()
 		case <-timer.C:
 		}
+		r.requestTelemetry(ctx, []attribute.KeyValue{
+			telemetry.WithReason(telemetry.StatusOverLimit),
+			telemetry.WithDecision("throttled"),
+			telemetry.WithThrottleBehavior("delay"),
```

Comment:
I dont see the use of throttle behaviour and can be removed, WDYT?

Reply:
Agree! Also the over limit reason needs to be removed too.